src/OpenTelemetry.AutoInstrumentation/Instrumentations/NLog/Bridge/Integrations/LoggerIntegration.cs

```diff
@@ -0,0 +1,109 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+using System.Diagnostics;
+using System.Reflection;
+using OpenTelemetry.AutoInstrumentation.CallTarget;
+using OpenTelemetry.AutoInstrumentation.Instrumentations.NLog.AutoInjection;
+using OpenTelemetry.AutoInstrumentation.Logging;
+#if NET
+using OpenTelemetry.AutoInstrumentation.Logger;
+#endif
+
+namespace OpenTelemetry.AutoInstrumentation.Instrumentations.NLog.Bridge.Integrations;
+
+/// <summary>
+/// NLog Logger integration that hooks into the actual logging process.
+/// This integration intercepts NLog's Logger.Log method calls to automatically
+/// capture log events and forward them to OpenTelemetry when the NLog bridge is enabled.
+///
+/// The integration targets NLog.Logger.Log method which is the core method called
+/// for all logging operations, allowing us to capture events without modifying configuration.
+/// </summary>
+[InstrumentMethod(
+assemblyName: "NLog",
+typeName: "NLog.Logger",
+methodName: "Log",
+returnTypeName: ClrNames.Void,
+parameterTypeNames: new[] { "NLog.LogEventInfo" },
```

Comment:
I think that you need to revisit the way how you injecting bytecode.

[`Logger.log(LogEventInfo)`](https://github.com/NLog/NLog/blob/809ef9cd79e76c6c53c6cccbd5778bd48579fdd0/src/NLog/Logger.cs#L268-L279) is not covering all cases when you are trying to log.

Even when I have checked the testing application calling this method, it is failing.